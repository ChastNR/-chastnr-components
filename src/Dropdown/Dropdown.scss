.dd {
  cursor: pointer;
  position: relative;
  display: inline-block;
  white-space: nowrap;
  max-width: 200px;

  .dd__cnt {
    align-items: center;
    display: flex;
    justify-content: center;
  }

  .dd__itms {
    display: none;
    border-radius: 4px;
    position: absolute;
    border: 1px solid #eaecee;
    box-shadow: 0 8px 16px rgba(51, 51, 51, 0.2);
    background-color: #fff;
    min-width: 100px;

    ul {
      display: flex;
      flex-direction: column;
      padding: 7px 0;
      margin: 0;
      list-style: none;
      width: 100%;

      li {
        color: rgba(0, 0, 0, 0.54);

        a {
          position: relative;
          display: flex;
          justify-content: space-between;
          box-sizing: border-box;
          color: inherit;
          text-decoration: none;
          cursor: pointer;
          outline: none;
          font-size: 13px;
          letter-spacing: 0;
          padding: 8px 15px 8px 16px;
          word-wrap: break-word;
          line-height: 16px;
          font-weight: 400;

          &:hover {
            background-color: #f6f7f8;
            color: #1464cc;
            outline: 0;
          }
        }
      }
    }
  }

  &:hover {
    .dd__itms {
      display: block;
    }
  }
}
