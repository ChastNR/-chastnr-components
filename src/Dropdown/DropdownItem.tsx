import { Link } from 'react-router-dom';

export interface IDropdownItem {
  id: string | number;
  title: React.ReactNode;
  to?: string;
  onClick?: React.MouseEventHandler<HTMLAnchorElement>;
  onKeyDown?: React.KeyboardEventHandler<HTMLAnchorElement>;
}

export const DropdownElement: React.FC<IDropdownItem> = ({ to, title, onClick, onKeyDown }) => (
  <li>
    <Link onClick={onClick} onKeyDown={onKeyDown} to={to}>
      {title}
    </Link>
  </li>
);
