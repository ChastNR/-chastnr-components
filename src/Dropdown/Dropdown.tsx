import { PropsWithChildren } from 'react';
import classNames from 'clsx';
import { DropdownElement, IDropdownItem } from './DropdownItem';

import './Dropdown.scss';

interface IDropdownProps
  extends PropsWithChildren<{
    className?: string;
    items: IDropdownItem[];
  }> {}

export const Dropdown: React.FC<IDropdownProps> = ({ className, children, items }) => {
  const dropdownClassNames = classNames('dd', className);

  return (
    <div className={dropdownClassNames}>
      <span className="dd__cnt">{children}</span>
      <span className="dd__itms">
        <ul>
          {items.map((i) => (
            <DropdownElement key={i.id} {...i} />
          ))}
        </ul>
      </span>
    </div>
  );
};
