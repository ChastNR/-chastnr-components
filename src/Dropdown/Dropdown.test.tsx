import { MemoryRouter } from "react-router-dom";

import { mount, shallow } from "enzyme";
import toJson from "enzyme-to-json";

import { DropdownItem } from "./DropdownItem";
import Dropdown from "./Dropdown";

describe("<Dropdown />", () => {
  const items: DropdownItem[] = [
    {
      id: 0,
      title: "testTitle",
      to: "/",
    },
  ];

  it("shoud equal snapshot", () => {
    const dropdown = shallow(<Dropdown items={items} />);
    expect(toJson(dropdown)).toMatchSnapshot();
  });

  it("props should be equal", () => {
    const dropdown = mount(
      <MemoryRouter>
        <Dropdown items={items} />
      </MemoryRouter>
    );
    const link = dropdown.find("a");

    expect(link.props().href).toEqual("/");
    expect(link.text()).toContain("testTitle");
  });
});
