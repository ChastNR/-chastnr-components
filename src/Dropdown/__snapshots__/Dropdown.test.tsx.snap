// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dropdown /> should match snapshot 1`] = `
<div
  class="dd"
>
  <span
    class="dd__cnt"
  />
  <span
    class="dd__itms"
  >
    <ul>
      <li>
        <a
          href="/"
        >
          testTitle
        </a>
      </li>
    </ul>
  </span>
</div>
`;
